BJS2-66534-like-notification-listener

gulnaz21 opened the pull request.

gulnaz21 committed:
BJS2-66534-like-notification-listener

marsel-mkh commented on src/main/java/faang/school/notificationservice/listener/LikeEventListener.java:
```diff
@@ -0,0 +1,27 @@
+    private final NotificationServiceSelector notificationServiceSelector;
+
+    public void handleMessage(LikePostEvent event) {
+        long postAuthorId = event.getPostAuthorId();
```
Можно добавить немного логирования

marsel-mkh commented on src/main/java/faang/school/notificationservice/listener/LikeEventListener.java:
```diff
@@ -0,0 +1,27 @@
+import java.util.Locale;
+
+@Component
+@RequiredArgsConstructor
```
добавь доку на класс

gulnaz21 committed:
fix:
  - добавлена JavaDoc
  - добавлено логирование

RinatBeybutov commented on src/main/java/faang/school/notificationservice/messaging/LikeMessageBuilder.java:
```diff
@@ -0,0 +1,50 @@
+     */
+    @Override
+    public String buildMessage(LikePostEvent event, Locale locale) {
+        UserDto postAuthor = userServiceClient.getUser(event.getPostAuthorId());
```
в LikeEventListener идет вызов на получение автора и здесь, дублирование получается. Нужно подумать, как это исправить

RinatBeybutov commented on src/main/java/faang/school/notificationservice/service/NotificationServiceSelector.java:
```diff
@@ -0,0 +1,31 @@
+     * @param message текст уведомления для отправки
+     */
+    public void notifyUser(UserDto user, String message) {
+        notificationServices.stream()
```
Владимир пишет абстрактный слушатель, его код почти. Нужно уточнить у него, здесь может переиспользовать его код

RinatBeybutov commented on src/main/resources/messages.properties:
```diff
@@ -1,2 +1,3 @@
-follower.new = "Congrats! You've got a new follower!"
\ No newline at end of file
+follower.new = "Congrats! You've got a new follower!"
+postLike.new = Congrats {0} \uD83C\uDF89 You have got a new like on your post from \"{1}\"! \uD83D\uDE0D
```
можно добавить в конец ключа число аргументов в тексте, те:
postLike.new.2, 
Это опционально

gulnaz21 committed:
Merge branch 'murloc-master-stream9' into feature/BJS2-66534-like-notification-listener

# Conflicts:
#	src/main/java/faang/school/notificationservice/dto/UserDto.java

gulnaz21 committed:
fix:
  - код переделан под созданный класс AbstractEventListener
  - в messages.properties название текста для лайка изменено с учетом кол-ва аргументов
  - изменен UserDto для корректной работы

gulnaz21 committed:
Merge branch 'murloc-master-stream9' into feature/BJS2-66534-like-notification-listener

# Conflicts:
#	src/main/java/faang/school/notificationservice/dto/UserDto.java
#	src/main/resources/application.yaml
#	src/main/resources/messages.properties

gulnaz21 committed:
fix:
  - решены конфликты
